### ISSUE TITLE
`@JsonIgnoreProperties` not working with `@JsonValue`

### ISSUE THREAD

**Issue by ThatSneakyRaccoon** [FasterXML/jackson-databind#3647]
When serializing the class `Bar` using the default `ObjectMapper`
```
class Foo {
     
     String p1 = "hello";
     String p2 = "world";
     
}
```
```
class Bar {

     @JsonValue
     @JsonIgnoreProperties( "p1" )
     Foo foo = new Foo();
    
}
```
The output is
```
{
  "p1" : "hello",
  "p2" : "world"
}
```
I would expect it to be
```
{
  "p2" : "world"
}
```
Tested on version **2.13.1**

**Comment by cowtowncoder** [comment]
Correct: this is a limitation and combination does not work.

I will leave this open as possible future enhancement if anyone has time and interest: I agree that it would make sense to support. But there is no on-going work in this area that I know of.

**Comment by mukham12** [comment]
Hi @cowtowncoder. 

I'm currently looking into it, but I'm still familiarizing myself with the codebase and haven't found the best approach yet. Obviously, handling fields marked with both @JsonValue and @JsonIgnoreProperties separately and together require different code flows, so I could not find an overlapping method that could be used to deal with the combo.

If you have any suggestions on where I should focus on speeding up development, I'd love to hear them! I'm open to feedback and appreciate any guidance you can provide. 

One of the challenges I'm facing is that, when serializing individual fields like `p1` and `p2`, we don't have a way to know if the class they are in (e.g. Foo) is marked with @JsonIgnoreProperties inside of Bar. This makes it more difficult to determine the correct behavior for each field. Or maybe I am just looking in the wrong place :)

Looking forward to hearing from you.

**Comment by cowtowncoder** [comment]
@mukham12 I don't remember off-hand exactly where to look (if this was simple to do I might have worked on it earlier :) ).

But there are 2 obvious things to try to look for (apologies if this is too obvious):

1. Handling of `JsonValueSerializer` (src/main/java/com/fasterxml/jackson/databind/ser/std/JsonValueSerializer.java) -- how it get constructed, configured etc
2. Handling of property-defined `@JsonIgnoreProperties`, how it propagated to POJOs (`BeanSerializer`, `BeanSerializerBase`) -- this is different from (but combined with) annotation as used for class itself

the trick, basically, would be to combine 2 things, to make `JsonValueSerializer` somehow pass configuration/contextualization of "ignorable property names" to `BeanSerializer`, but ideally in a way that does not hard-code type (but perhaps it has to; they are ways around this like tag-interfaces).

**Comment by cowtowncoder** [comment]
Also, one other minor note: writing a (failing) unit test, to be added under `src/test/java/com/fasterxml/jackson/failing` would be useful as well; this is often the starting point for implementations.

**Comment by mukham12** [comment]
Thanks a lot for the suggestions. I will keep looking into this and I am going to keep you posted.

**Comment by cowtowncoder** [comment]
Forgot to close; included in 2.16.0-rc1.
